packages/cms-data-sync/src/news/news.data-migration.ts

```diff
@@ -0,0 +1,110 @@
+/* eslint-disable no-console */
+/* eslint-disable no-restricted-syntax */
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+
+import { Entry, SysLink } from 'contentful-management';
+import { Document } from '@contentful/rich-text-types';
+
+import { newsQuery } from './news.queries';
+import {
+  FetchNewsQuery,
+  FetchNewsQueryVariables,
+} from '../autogenerated-gql/graphql';
+
+import {
+  clearContentfulEntries,
+  convertHtmlToContentfulFormat,
+  createAssetLink,
+  getSquidexAndContentfulClients,
+  publishContentfulEntries,
+} from '../utils';
+
+type NewsPayload = {
```

Comment:
although it may sound like an overkill i think we should be able to have these types generated in the contentful package. We use graphql ones there at the moment cause we decided it was easier but i think they provide tools for working with their client libraries? If we use the generated types it will be less error prone, especially since we don't do any integration tests for this migration so types is the only real test available

Reply:
I've just tried this lib https://github.com/intercom/contentful-typescript-codegen

and it generated this

```
// THIS FILE IS AUTOMATICALLY GENERATED. DO NOT MODIFY IT.

import { Asset, Entry } from 'contentful';
import { Document } from '@contentful/rich-text-types';

export interface IMigrationFields {
  /** Migration State */
  state: Record<string, any>;

  /** Content Type ID */
  contentTypeId: string;
}

/** Meta data to store the state of content model through migrations */

export interface IMigration extends Entry<IMigrationFields> {
  sys: {
    id: string;
    type: string;
    createdAt: string;
    updatedAt: string;
    locale: string;
    contentType: {
      sys: {
        id: 'migration';
        linkType: 'ContentType';
        type: 'Link';
      };
    };
  };
}

export interface INewsFields {
  /** Title */
  title: string;

  /** Short Text */
  shortText?: string | undefined;

  /** Thumbnail */
  thumbnail?: Asset | undefined;

  /** Frequency */
  frequency: 'Biweekly Newsletter' | 'CRN Quarterly' | 'News Articles';

  /** External Link */
  link?: string | undefined;

  /** External Link Text */
  linkText?: string | undefined;

  /** Text */
  text?: Document | undefined;
}

/** ASAP Hub News */

export interface INews extends Entry<INewsFields> {
  sys: {
    id: string;
    type: string;
    createdAt: string;
    updatedAt: string;
    locale: string;
    contentType: {
      sys: {
        id: 'news';
        linkType: 'ContentType';
        type: 'Link';
      };
    };
  };
}

export type CONTENT_TYPE = 'migration' | 'news';

export type IEntry = IMigration | INews;

export type LOCALE_CODE = 'en-US';

export type CONTENTFUL_DEFAULT_LOCALE_CODE = 'en-US';
```

It's not like the payload for creating an entry but what comes back from the API. I google it for some time but I could not find another lib that does that. Since it doesn't have the types that we need, I won't commit that, is that ok?

Reply:
yeah its fine i thought it would generate the input. Weird